### PR TITLE
DAOS-625 test: fix test typo

### PR DESCRIPTION
**Pull request by mchaarawi**
Quick-Functional: true
Test-tag: test_daos_dfs_unit

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/tests/suite/dfs_unit_test.c [hunk 1]
```diff
@@ -1336,7 +1336,7 @@ dfs_test_chown(void **state)
 	rc = dfs_stat(dfs_mt, NULL, filename_file2, &stbuf);
 	assert_int_equal(rc, 0);
 	assert_int_equal(stbuf.st_uid, stbuf2.st_uid);
-	assert_int_equal(stbuf.st_uid, stbuf2.st_gid);
+	assert_int_equal(stbuf.st_gid, stbuf2.st_gid);
```
Comment (ashleypittman) [inline src/tests/suite/dfs_unit_test.c:1339]: How did this pass before?

Comment (mchaarawi) [reply src/tests/suite/dfs_unit_test.c:1339]: probably in CI uid == gid

**Comment by mchaarawi** [comment]
test passed:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-9427/1/artifact/Functional%20Hardware%20Large/daos_test/dfs.py/job.log

File: src/tests/suite/dfs_unit_test.c (hunk 1 above)
Comment (mchaarawi) [reply src/tests/suite/dfs_unit_test.c:1339]: as we discussed, actually it was failing but it was not detected in CI for some reason. i opened:
https://daosio.atlassian.net/browse/DAOS-10882

**Review by ashleypittman — APPROVED** [review]
The fix looks good, but I do think we need to look seriously at why this wasn't caught in CI, and perhaps get a PR ready that would correctly fail with the code as is.

**Review by wiliamhuang — APPROVED** [review]

**Pull request merged**